ion_schema_2_0/constraints/regex.isl

```diff
@@ -0,0 +1,1091 @@
+    null,
+    "",
+    "\n",
+    "\r",
+  ]
+}
+
+type::{
+  name: regex_character_class_compound_negated,
+  regex: "[^ac-eh-jmnt-vz]",
+}
+$test::{
+  type: regex_character_class_compound_negated,
+  should_accept_as_valid: [
+    "b",
+    "f",
+    "g",
+    "k",
+    "l",
+    "o",
+    "p",
+    "q",
+    "r",
+    "s",
+    "w",
+    "x",
+    "y",
+  ],
+  should_reject_as_invalid: [
+    "a",
+    "c",
+    "d",
+    "e",
+    "h",
+    "i",
+    "j",
+    "m",
+    "n",
+    "t",
+    "u",
+    "v",
+    "z",
+  ]
+}
+
+type::{
+  name: regex_character_class_dot,
+  regex: "[.]",
+}
+$test::{
+  type: regex_character_class_dot,
+  should_accept_as_valid: [
+    '.',
+    ".",
+  ],
+  should_reject_as_invalid: [
+    'a',
+    "a",
+  ]
+}
+
+type::{
+  name: regex_character_class_escape_chars,
+  regex: "[\\[\\]\\\\]",
+}
+$test::{
+  type: regex_character_class_escape_chars,
+  should_accept_as_valid: [
+    "[",
+    "]",
+    "\\", //"// <-- This extra comment markers and quote is to stop intellij from messing up the code formatting. https://github.com/amzn/ion-intellij-plugin/issues/40
+  ],
+  should_reject_as_invalid: [
+    "a",
+    ".",
+  ]
+}
+
+type::{
+  name: regex_character_class_negated,
+  regex: "[^bcd]",
+}
+$test::{
+  type: regex_character_class_negated,
+  should_accept_as_valid: [
+    "a",
+    "e",
+  ],
+  should_reject_as_invalid: [
+    "b",
+    "c",
+    "d",
+  ]
+}
+
+type::{
+  name: regex_character_class_special_chars,
+  regex: "[\0\a\b\t\n\v\f\r]",
+}
+$test::{
+  type: regex_character_class_special_chars,
+  should_accept_as_valid: [
+    "\0",
+    "\a",
+    "\b",
+    "\t",
+    "\n",
+    "\v",
+    "\f",
+    "\r",
+  ],
+  should_reject_as_invalid: [
+    "a",
+    ".",
+  ]
+}
+
+type::{
+  name: regex_character_class_word_char,
+  regex: "[A-Za-z0-9]",
+}
+$test::{
+  type: regex_character_class_word_char,
+  should_accept_as_valid: [
+    "A",
+    "M",
+    "Z",
+    "a",
+    "n",
+    "z",
+    "0",
+    "5",
+    "9",
+  ],
+  should_reject_as_invalid: [
+    " ",
+    ".",
+    "\t",
+    "#",
+    "~",
+  ]
+}
+
+type::{
+  name: regex_character_class,
+  regex: "[bcd]",
+}
+$test::{
+  type: regex_character_class,
+  should_accept_as_valid: [
+    "b",
+    "c",
+    "d",
+  ],
+  should_reject_as_invalid: [
+    "a",
+    "e",
+  ]
+}
+
+type::{
+  name: regex_character_range_negated,
+  regex: "[^b-g]",
+}
+$test::{
+  type: regex_character_range_negated,
+  should_accept_as_valid: [
+    "a",
+    "i",
+  ],
+  should_reject_as_invalid: [
+    "b",
+    "c",
+    "d",
+    "e",
+    "f",
+    "g",
+  ]
+}
+
+type::{
+  name: regex_character_range,
+  regex: "[b-g]",
+}
+$test::{
+  type: regex_character_range,
+  should_accept_as_valid: [
+    "b",
+    "c",
+    "d",
+    "e",
+    "f",
+    "g",
+  ],
+  should_reject_as_invalid: [
+    "a",
+    "i",
+  ]
+}
+
+type::{
+  name: regex_complex_nested_quantified_alternation,
+  regex: "^(ab|(cd){2,3}|ef){2}$",
+}
+$test::{
```

Comment:
Most of these test cases were copied directly from the Ion Schema 1.0 tests since there is very little that is changing for regex between Ion Schema 1.0 and 2.0.

I'll go back and see about making these tests simpler.